File: web-demo/test/test_encoder_minimal.js
#!/usr/bin/env node
/*
 * Minimal encoder test - load the same input as Python and compare
 */

const fs = require('fs');
const path = require('path');
const ort = require('onnxruntime-node');

async function main() {
    // Load the transposed features from JS
    const jsFeatures = JSON.parse(fs.readFileSync('js_features.json', 'utf-8'));

    // Create transposed array like Python
    const T = jsFeatures.shape[0];  // 82
    const featDim = jsFeatures.shape[1];  // 37
    const transposed = new Float32Array(featDim * T);

    for (let t = 0; t < T; t++) {
        for (let f = 0; f < featDim; f++) {
            transposed[f * T + t] = jsFeatures.data[t * featDim + f];
        }
    }

    console.log(`Input shape: [1, ${featDim}, ${T}]`);
    console.log(`First 10 values: [${transposed.slice(0, 10).map(x => x.toFixed(6)).join(', ')}]`);

    // Load and run encoder with explicit CPU provider
    const encoderPath = path.join(__dirname, '..', 'models', 'correct_9292025', 'encoder.onnx');
    const encoderSession = await ort.InferenceSession.create(encoderPath, {
        executionProviders: ['cpu']
    });

    const encInputs = {
        'audio_signal': new ort.Tensor('float32', transposed, [1, featDim, T]),
        'length': new ort.Tensor('int64', BigInt64Array.from([BigInt(T)]), [1])
    };

    const encOut = await encoderSession.run(encInputs);
    const encoded = encOut['outputs'];
    const encodedLen = Number(encOut['encoded_lengths'].data[0]);

    console.log(`Encoder output shape: [${encoded.dims}]`);
    console.log(`Encoded length: ${encodedLen}`);
    console.log(`First 10 values: [${encoded.data.slice(0, 10).map(x => x.toFixed(6)).join(', ')}]`);

    // Save output for comparison
    const outputData = {
        shape: encoded.dims,
        data: Array.from(encoded.data)
    };
    fs.writeFileSync('encoder_output_js.json', JSON.stringify(outputData, null, 2));
    console.log('Saved encoder output to encoder_output_js.json');
}

main().catch(console.error);

File: web-demo/test/test_encoder_minimal.py
#!/usr/bin/env python3
"""
Minimal encoder test - save input and output for comparison
"""

import numpy as np
import onnxruntime as ort
import json

# Load features
features = np.load('python_features.npy')
print(f'Features shape: {features.shape}')

# Transpose for encoder
signal = features.T.reshape(1, 37, -1).astype(np.float32)
signal_len = np.array([82], dtype=np.int64)

# Save input for comparison
np.save('encoder_input.npy', signal)
print(f'Saved encoder input to encoder_input.npy')

# Load and run encoder
encoder = ort.InferenceSession('../models/correct_9292025/encoder.onnx')
enc_out = encoder.run(None, {'audio_signal': signal, 'length': signal_len})

encoded = enc_out[0]
encoded_len = enc_out[1]

print(f'Encoder output shape: {encoded.shape}')
print(f'Encoded length: {encoded_len}')
print(f'First 10 values: {encoded.flat[:10]}')

# Save output
np.save('encoder_output_python.npy', encoded)
print(f'Saved encoder output to encoder_output_python.npy')